artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ReplicationBackupActivation.java

```diff
@@ -0,0 +1,489 @@
+         } else {
+            ActiveMQServerLogger.LOGGER.serverIsLive();
+         }
+         activeMQServer.completeActivation(true);
+      }
+   }
+
+   private LiveNodeLocator createLiveNodeLocator(final LiveNodeLocator.BackupRegistrationListener registrationListener) {
+      if (expectedNodeID != null) {
+         assert failback;
+         return new NamedLiveNodeIdLocatorForReplication(expectedNodeID, registrationListener, policy.getRetryReplicationWait());
+      }
+      return policy.getGroupName() == null ?
+         new AnyLiveNodeLocatorForReplication(registrationListener, activeMQServer, policy.getRetryReplicationWait()) :
+         new NamedLiveNodeLocatorForReplication(policy.getGroupName(), registrationListener, policy.getRetryReplicationWait());
+   }
+
+   private DistributedLock replicateAndFailover(final ClusterController clusterController) throws ActiveMQException, InterruptedException {
+      while (true) {
+         synchronized (this) {
+            if (closed) {
+               return null;
+            }
+         }
+         final ReplicationFailure failure = replicateLive(clusterController);
+         if (failure == null) {
+            Thread.sleep(clusterController.getRetryIntervalForReplicatedCluster());
+            continue;
+         }
+         if (!activeMQServer.isStarted()) {
+            return null;
+         }
+         LOGGER.debugf("ReplicationFailure = %s", failure);
+         switch (failure) {
+            case VoluntaryFailOver:
+            case NonVoluntaryFailover:
+               final DistributedLock liveLock = tryAcquireLiveLock();
+               if (liveLock != null) {
+                  return liveLock;
+               }
+               if (!failback) {
+                  ActiveMQServerLogger.LOGGER.restartingAsBackupBasedOnQuorumVoteResults();
+               }
+               asyncRestartServer(activeMQServer, true);
+               return null;
+            case RegistrationError:
+               asyncRestartServer(activeMQServer, false);
+               return null;
+            case AlreadyReplicating:
+               // can just retry here, data should be clean
+               continue;
+            case ClosedObserver:
+               return null;
+            case BackupNotInSync:
+               asyncRestartServer(activeMQServer, true);
+               return null;
+            case WrongNodeId:
+               asyncRestartServer(activeMQServer, true);
+               return null;
+            default:
+               throw new AssertionError("Unsupported failure " + failure);
+         }
+      }
+   }
+
+   private DistributedLock tryAcquireLiveLock() throws InterruptedException {
+      assert activeMQServer.getNodeManager().getNodeId() != null;
+      final String liveID = activeMQServer.getNodeManager().getNodeId().toString();
+      final int voteRetries = policy.getVoteRetries();
+      final long maxAttempts = voteRetries >= 0 ? (voteRetries + 1) : -1;
```

Comment:
@michaelandrepearce
@jbertram 
 It seems that `vote-retries` can be set to be `-1` (unbounded) and that it was used with the meaning that `vote-retries` == 0 is still an option...
I've decided to change the semantic to better match its name ie "retries" it seems to me (I'm not native eng, so please correct me if I'm wrong eh!) that are "additional" to the first mandatory "try".

Reply:
so 

-1 normally means to me "unlimited" e.g. never stop

0 means dont do any retry, e.g. fail/error on first error, and do not re-attempt.

Reply:
in regards to replication, i would typically expect.

-1 normally means wait until all members have acked.
0 means do not wait
1 would wait till 1 member acked.

Reply:
Not sure about the "members" count: I believe it's the amount of retrying to get the majority of votes (or the majority of the maximum seen members? Not sure) that would agree that the previous live is down for real - for the "old" replication. 
Here the semantic is similar, but how a completed/successfull vote is going to happen is an implementation detail of the "plugged" quorum vote algorithm aka the distributed lock and it's not just a check on the existing live role, but an atomic check-then-act to allow this backup to start failover too.
In order to make this a bit more resistant I have decided that both a failed quorum vote due to loss of quorum and a failed attempt to become live (because the existing live is still stopping or it's an impl detail of the quorum vote provider eg raft/zk leader re-election) are seen as retryable instead of failing fast. To fail fast the user must set the retries to 0.
According to the existing failover test this seems to play nice with the existing user experience.

Reply:
I see this doc related `vote-retries`:

> vote-retries
This property controls how many times the backup broker retries the quorum vote in order to receive a majority result that allows the backup broker to start up.